README.md

```diff
@@ -149,6 +149,62 @@ client.chat(
 # => "Anna is a young woman in her mid-twenties, with wavy chestnut hair that falls to her shoulders..."
 ```
 
+### Function Calling
+
+Developers can describe functions and have the model intelligently choose to output a JSON object containing arguments to call those functions.
+
+```ruby
+def get_current_weather(location:, unit:)
+  # use a weather api to fetch weather
+end
+
+response =
+  client.chat(
+    parameters: {
+      model: MODEL_NAME,
+      messages: [
+        {
+          "role": "user",
+          "content": "What is the weather like in San Francisco?",
+        },
+      ],
+      functions: [
```

Comment:
does this work?

Reply:
Yes —

"What is the weather like in San Francisco?" returns:
```ruby
{"role"=>"assistant", "content"=>nil, "function_call"=>{"name"=>"get_current_weather", "arguments"=>"{\n  \"location\": \"San Francisco, CA\"\n}"}}
```

"What is the weather like in San Francisco in celsius?" returns:
```ruby
{"role"=>"assistant", "content"=>nil, "function_call"=>{"name"=>"get_current_weather", "arguments"=>"{\n  \"location\": \"San Francisco, CA\",\n  \"unit\": \"celsius\"\n}"}}
```

"What is 1+1" returns:
```ruby
{"role"=>"assistant", "content"=>"1 + 1 is equal to 2."}
```

Reply:
Hmm, when I run this I just get:
```
       {"message"=>"Unrecognized request argument supplied: functions", "type"=>"invalid_request_error", "param"=>nil, "code"=>nil}```

Reply:
Ah i get it. You need to use the latest model (for example): 

`gpt-4-0613`